src/qutip_qip/circuit/mat_renderer.py

```diff
@@ -0,0 +1,1025 @@
+
+        bbox = text_obj.get_window_extent(
+            renderer=self._ax.figure.canvas.get_renderer()
+        )
+        inv = self._ax.transData.inverted()
+        bbox_data = bbox.transformed(inv)
+        text_obj.remove()
+
+        return bbox_data.width * 2.54 * 3
+
+    def _manage_layers(
+        self,
+        gate_width: float,
+        wire_list: List[int],
+        layer: int,
+        xskip: float = 0,
+    ) -> None:
+        """
+        Manages and updates the layer widths according to the gate's width just plotted.
+
+        Parameters
+        ----------
+        gate_width : float
+            The width of the gate to be plotted.
+
+        wire_list : list
+            The list of wires the gate is acting on (control and target).
+
+        layer : int
+            The layer the gate is acting on.
+
+        xskip : float, optional
+            The horizontal value for getting to requested layer. The default is 0.
+        """
+
+        for wire in wire_list:
+            if len(self._layer_list[wire]) > layer:
+                if (
+                    self._layer_list[wire][layer]
+                    < gate_width + self.style.gate_margin * 2
+                ):
+                    self._layer_list[wire][layer] = (
+                        gate_width + self.style.gate_margin * 2
+                    )
+            else:
+                temp = xskip - sum(self._layer_list[wire]) if xskip != 0 else 0
+                self._layer_list[wire].append(
+                    temp + gate_width + self.style.gate_margin * 2
+                )
+
+    def _add_wire(self) -> None:
+        """
+        Adds the wires to the circuit.
+        """
+        max_len = (
+            max([sum(self._layer_list[i]) for i in range(self._qwires)])
+            + self.style.end_wire_ext * self.style.layer_sep
+        )
+
+        for i in range(self._qwires):
+            wire = plt.Line2D(
+                [0, max_len],
+                [i * self.style.wire_sep, i * self.style.wire_sep],
+                lw=1,
+                color=self.style.wire_color,
+                zorder=self._zorder["wire"],
+            )
+            self._ax.add_line(wire)
+
+        for i in range(self._cwires):
+            wire_up = plt.Line2D(
+                [0, max_len],
+                [
+                    (i + self._qwires) * self.style.wire_sep,
+                    (i + self._qwires) * self.style.wire_sep,
+                ],
+                lw=1,
+                color=self.style.wire_color,
+                zorder=self._zorder["wire"],
+            )
+            wire_down = plt.Line2D(
+                [0, max_len],
+                [
+                    (i + self._qwires) * self.style.wire_sep,
+                    (i + self._qwires) * self.style.wire_sep,
+                ],
+                lw=1,
+                color=self.style.wire_color,
+                zorder=self._zorder["wire"],
+            )
```

Comment:
What is the difference between this part and the `_draw_qbridge` and `_draw_cbridge`?

Reply:
This part refers to rendering the classical wire horizontally, where as `_draw_cbridge` and `_draw_qbridge` are to renderer classical and qubit across the two wires. Example, `_draw_cbridge` is used to render the connection from a mesurment gate to classical line and  `_draw_qbridge` are used for connecting control node to gates.